connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConnector.java

```diff
@@ -308,15 +308,16 @@ void doShutdown() {
                                     + " as the connector has been scheduled for shutdown"),
                         null);
             }
-            if (state == State.STARTED)
+            // Call stop() when the connector has failed, for the connector
+            // to release its resources.
+            if (state == State.STARTED || state == State.FAILED)
```

Comment:
@ashoke-cube , can we also add `state == State.INIT` as a condition here? There could be cases where the `connector.stop()` method takes a long time and in the meanwhile. As we can see [here](https://github.com/apache/kafka/blob/ceec218351ff741dc338edf8d3810c1f36555e33/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConnector.java#L213-L218), the connector won't transition to `STARTED` state till the method doesn't return. In such cases, if someone tries to delete the connector, the resources won't be released because the flow won't enter this condition. 
If we want to keep this change local to startup failure, we can do that as well. Let me know.